lib/plugins/unthreaded-signing/plugin.c

```diff
@@ -88,6 +88,8 @@ unthreaded_openssl_sign_hash(sv_unthreaded_plugin_t *self, const uint8_t *hash,
   int idx = self->out_buffer_idx;
   if (idx < MAX_BUFFER_LENGTH) {
     status = openssl_sign_hash(&self->sign_data);
+  } else {
+    return SV_NOT_SUPPORTED;
```

Comment:
You mix a lot of returns wrapped in if-else-statements which makes it hard to read and understand what actually happens. I would have written this something like

```
int idx = self->out_buffer_idx;
if (idx >= MAX_BUFFER_LENGTH) {
  status = SV_NOT_SUPPORTED;
  goto done;
}

status = openssl_sign_hash(&self->sign_data);
if (status != SV_OK) goto done;

if (self->sign_data.signature_size == 0) {
  status = SV_NOT_SUPPORTED;
  goto done;
}

signature_data_t *out = &self->out_buffer[idx];
if (!out->signature) {
  out->signature = calloc(1, self->sign_data.max_signature_size);
  if (!out->signature) {
    status = SV_MEMORY;
    goto done;
  }
}

memcpy(out->signature, self->sign_data.signature, self->sign_data.signature_size);
out->signature_size = self->sign_data.signature_size;
self->out_buffer_idx++;

done:
return status;
```